n2kMapper.js

```diff
@@ -4,9 +4,16 @@ var through = require('through');
 var debug = require('debug')('signalk:n2k-signalk')
 
 
-var toDelta = function(n2k) {
+var toDelta = function(n2k, state) {
   try {
     var theMappings = n2kMappings[n2k.pgn];
+    if ( state ) {
+      var n2k_src = n2k.src.toString()
+      if ( ! state[n2k_src] ) {
+        state[n2k_src] = {}
+      }
+      state = state[n2k_src]
```

Comment:
I think it is bad practice to reassign function parameters - the state passed in and the per source state should be different variables.